src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/BatchExecutionTest.java

```diff
@@ -167,6 +230,61 @@ public void testExecuteBatch1() {
             fail(TestResource.getResource("R_executeBatchFailed") + ": " + e.getMessage());
         }
     }
+    
+    public void testExecuteBatch1UseBulkCopyAPI() {
+        int i = 0;
+        int retValue[] = {0, 0, 0};
+        int updCountLength = 0;
+        try {
+            String sPrepStmt = "update ctstable2 set PRICE=PRICE*20 where TYPE_ID=?";
+
+            pstmt = connection.prepareStatement(sPrepStmt);
+            Field f1 = SQLServerConnection.class.getDeclaredField("isAzureDW");
+            f1.setAccessible(true);
+            f1.set(connection, true);
+            pstmt.setInt(1, 1);
+            pstmt.addBatch();
+
+            pstmt.setInt(1, 2);
+            pstmt.addBatch();
+
+            pstmt.setInt(1, 3);
+            pstmt.addBatch();
+
+            int[] updateCount = pstmt.executeBatch();
+            updCountLength = updateCount.length;
+
+            assertTrue(updCountLength == 3, "executeBatch does not execute the Batch of SQL statements, Call to executeBatch is Failed!");
+
+            String sPrepStmt1 = "select count(*) from ctstable2 where TYPE_ID=?";
+
+            pstmt1 = connection.prepareStatement(sPrepStmt1);
+
+            for (int n = 1; n <= 3; n++) {
+                pstmt1.setInt(1, n);
+                rs = pstmt1.executeQuery();
+                rs.next();
+                retValue[i++] = rs.getInt(1);
+            }
+
+            pstmt1.close();
+
+            for (int j = 0; j < updateCount.length; j++) {
+                if (updateCount[j] != retValue[j] && updateCount[j] != Statement.SUCCESS_NO_INFO) {
+                    fail("executeBatch does not execute the Batch of SQL statements, Call to executeBatch is Failed!");
+                }
+            }
+        }
+        catch (BatchUpdateException b) {
+            fail("BatchUpdateException :  Call to executeBatch is Failed!");
+        }
+        catch (SQLException sqle) {
+            fail("Call to executeBatch is Failed!");
```

Comment:
These catch blocks make no sense.

Reply:
I agree that they could handle exception better here, but this isn't my code. I'm just gonna fix it anyways though.